website/build_javadocs.sh

```diff
@@ -0,0 +1,181 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+# Find project root directory
+SCRIPT_PATH=$(cd "$(dirname "$0")" && pwd)
+cd "$SCRIPT_PATH"
+
+# Navigate to project root by looking for pom.xml
+while [[ ! -f "pom.xml" && "$(pwd)" != "/" ]]; do
+  cd ..
+done
+
+# Validate we found the project root
+if [[ ! -f "pom.xml" ]]; then
+  echo "Error: Could not find project root with pom.xml"
+  exit 1
+fi
+
+echo "Found project root: $(pwd)"
+
+# Extract version from Maven project
+VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout 2>/dev/null || mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+if [[ -z "$VERSION" ]]; then
+  echo "Error: Could not extract project version"
+  exit 1
+fi
+
+# Get current branch name for version determination
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD 2>/dev/null || echo "unknown")
+echo "Current branch: $CURRENT_BRANCH"
+
+# Determine javadoc version based on branch type
+if [[ "$CURRENT_BRANCH" == "main" ]]; then
+  JAVADOC_VERSION="main"
+elif [[ "$CURRENT_BRANCH" =~ ^release-([0-9]+\.[0-9]+)$ ]]; then
+  JAVADOC_VERSION="${BASH_REMATCH[1]}"
+else
+  # For feature branches or other branches, use sanitized branch name
+  JAVADOC_VERSION=$(echo "$CURRENT_BRANCH" | sed 's/[^a-zA-Z0-9._-]/_/g')
+fi
+
+# Setup output directory
+OUTPUT_DIR="website/static/javadoc/$JAVADOC_VERSION"
+echo "Generating Javadoc for version: $JAVADOC_VERSION"
+echo "Output directory: $OUTPUT_DIR"
+
+# Create output directory
+rm -rf "$OUTPUT_DIR"
+mkdir -p "$OUTPUT_DIR"
+
+# Create temporary POM for targeted javadoc generation
+TEMP_POM="temp-javadoc-pom.xml"
+cat > "$TEMP_POM" << EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.fluss</groupId>
+    <artifactId>fluss-javadoc-temp</artifactId>
+    <version>$VERSION</version>
+    <packaging>pom</packaging>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <modules>
+        <module>fluss-client</module>
+        <module>fluss-flink/fluss-flink-common</module>
+    </modules>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.6.3</version>
+                <configuration>
+                    <windowTitle>Apache Fluss \${project.version} API</windowTitle>
+                    <doctitle>Apache Fluss \${project.version} API</doctitle>
```

Comment:
If we move the changes to root pom, we can use `build-helper-maven-plugin` to parse project version into without `-SNAPSHOT` suffix (e.g., the following `base.version` property). 


```xml
<build>
  <plugins>
    <plugin>
      <groupId>org.codehaus.mojo</groupId>
      <artifactId>build-helper-maven-plugin</artifactId>
      <version>3.5.0</version>
      <executions>
        <execution>
          <id>parse-version</id>
          <goals>
            <goal>parse-version</goal>
          </goals>
        </execution>
      </executions>
    </plugin>
  </plugins>
</build>
```

```xml
<properties>
  <base.version>${parsedVersion.majorVersion}.${parsedVersion.minorVersion}</base.version>
</properties>
```